packages/signer/keyring-eth/src/internal/app-binder/task/SendSignPersonalMessageTask.ts

```diff
@@ -49,25 +49,24 @@ export class SendSignPersonalMessageTask {
 
     const buffer = builder.build();
 
-    let result = CommandResultFactory<SignPersonalMessageCommandResponse, void>(
-      { data: Nothing },
-    );
-
-    // Split the buffer into chunks
-    for (let i = 0; i < buffer.length; i += APDU_MAX_PAYLOAD) {
-      result = await this.api.sendCommand(
-        new SignPersonalMessageCommand({
-          data: buffer.slice(i, i + APDU_MAX_PAYLOAD),
-          isFirstChunk: i === 0,
-        }),
-      );
+    const result =
+      await new SendCommandInChunksTask<SignPersonalMessageCommandResponse>(
+        this.api,
+        {
+          data: buffer,
+          commandFactory: (args) =>
+            new SignPersonalMessageCommand({
+              data: args.chunkedData,
+              isFirstChunk: args.isFirstChunk,
+            }),
+        },
+      ).run();
 
-      if (!isSuccessCommandResult(result)) {
-        return result;
-      }
+    if (!isSuccessCommandResult(result)) {
+      return result;
     }
 
-    if (isSuccessCommandResult(result) && result.data.isJust()) {
+    if (result.data.isJust()) {
```

Comment:
Can you please test this snippet, I think that we should take the advantage of making chaining methods. 
```suggestion
    return result.data
      .ifJust(data => CommandResultFactory({ data }))
      .ifNothing(() => CommandResultFactory({
        error: new InvalidStatusWordError("no signature returned"),	
      }));
```

Reply:
This chaining code is not working, because we need to return the commandFactory,

but we can use a `mapOrDefault`operator instead:

```ts
return result.data.mapOrDefault(
      (data) => CommandResultFactory({ data }),
      CommandResultFactory({
        error: new InvalidStatusWordError("no signature returned"),
      }),
    );
```

Reply:
Yes, you're right. The `ifXXXX` method always return `this`. But let's make a chain. 😄